### ISSUE TITLE
cron module bugs

### ISSUE THREAD

**Issue by dirtyharrycallahan** [ansible/ansible#3304]
There is a defect in the cron module when running on SuSE and other distributions that add a header, three lines, to the user crontab file. Each call to add/remove a job includes the previous header. After adding the code to strip the header, I created a test suite outside ansible and found other problems with the handling empty jobs and cron files. I should be able to cleanup my changes and generate a pull request over the weekend.

**Comment by mpdehaan** [comment]
Yes, in the least, please write up what the other problems are too, and how to reproduce them.

Thanks!

**Comment by mpdehaan** [comment]
Any update on this one?

Note for future:  another user mentioned that setting up hour=5,2 runs on every minute, so perhaps the minutes should default (and other fields).

**Comment by dirtyharrycallahan** [comment]
I have been on the road. I will try to finish this up over the weekend.

**Comment by dirtyharrycallahan** [comment]
I worked on refactoring cron over the weekend but I need to completing testing on other OSs. So far I have only tested on CentOS 6.4 and I need to test on SuSE b/c of  the new feature to strip the SuSE added header. One thing that came up in my testing was the handling of white space in params. In the case of cron there may be a name which is used as a comment and the current behavior, which I have retained, name="foo" is not the same as name=" foo" or name=" foo ". While I could trim strings when white space matters I wondered what if any thoughts there are on white-space.
